pkg/runtime/generate.go

```diff
@@ -0,0 +1,282 @@
+		kubePut.WriteString(a.names[i])
+		kubePut.WriteString(",\n")
+
+		// api_group
+		apiGroup := object.GetObjectKind().GroupVersionKind().Group
+		if apiGroup != "" {
+			kubePut.Write(indent2)
+			kubePut.WriteString("api_group=\"" + apiGroup + "\",\n")
+		}
+
+		// data
+		kubePut.Write(indent2)
+		kubePut.WriteString("data=[")
+		kubePut.Write(data)
+		kubePut.WriteString("]\n")
+
+		kubePut.Write(indent1)
+		kubePut.WriteString(")\n")
+
+		if i != len(a.objects)-1 {
+			kubePut.WriteString("\n")
+		}
+	}
+	return kubePut.Bytes()
+}
+
+func (a *addonFile) getKubeDelete(indent int) []byte {
+	indent1 := bytes.Repeat([]byte(indentString), indent)
+	kubeDelete := bytes.NewBuffer([]byte{})
+	for _, object := range a.objects {
+		v := reflect.ValueOf(object)
+		if v.Kind() == reflect.Ptr {
+			v = v.Elem()
+		}
+		typeMeta := v.FieldByName("TypeMeta")
+		objectMeta := v.FieldByName("ObjectMeta")
+		if typeMeta.IsZero() || objectMeta.IsZero() {
+			continue
+		}
+		kind := typeMeta.FieldByName("Kind")
+		name := objectMeta.FieldByName("Name")
+		namespace := objectMeta.FieldByName("Namespace")
+		apiGroup := object.GetObjectKind().GroupVersionKind().Group
+		if kind.IsZero() || name.IsZero() {
+			continue
+		}
+		kubeDelete.Write(indent1)
+		kubeDelete.WriteString("kube.delete(")
+		kubeDelete.WriteString(strings.ToLower(kind.String()))
+		kubeDelete.WriteString("=\"")
+		if !namespace.IsZero() {
+			kubeDelete.WriteString(namespace.String())
+			kubeDelete.WriteString("/")
+		}
+		kubeDelete.WriteString(name.String())
+		kubeDelete.WriteString("\"")
+		if apiGroup != "" {
+			kubeDelete.WriteString(fmt.Sprintf(", api_group=\"%s\"", apiGroup))
+		}
+		kubeDelete.WriteString(")\n")
+	}
+	return kubeDelete.Bytes()
+}
+
+func (a *addonFile) getData(v reflect.Value, indent int) []byte {
+	indent1 := bytes.Repeat([]byte(indentString), indent)
+	indentTopLevel := indent1
+	if indent > 0 {
+		indentTopLevel = bytes.Repeat([]byte(indentString), indent-1)
+	}
+
+	b := bytes.NewBuffer([]byte{})
+
+	if v.Kind() == reflect.Ptr {
+		v = v.Elem()
+	}
+	if v.Kind() != reflect.Struct {
+		j, _ := json.Marshal(v.Interface())
+		if bytes.Equal([]byte("true"), j) || bytes.Equal([]byte("false"), j) {
+			// because Python's boolean is capitalized
+			return bytes.Title(j)
+		}
+		return j
+	}
+
+	t := v.Type()
+	name, pkgPath := t.Name(), t.PkgPath()
+	alias := a.addPkg(strings.ReplaceAll(pkgPath, "/", "."))
+	b.WriteString(alias + "." + name + "(\n")
+
+	for i := 0; i < v.NumField(); i++ {
+		vf := v.Field(i)
+		if !vf.IsZero() {
+			jsonTag := t.Field(i).Tag.Get("json")
+			// this is even a slice of len 1, if jsonTag is "". So accessing 0 index is safe
+			jsonTag = strings.Split(jsonTag, ",")[0]
+
+			// add name in order for use in getKubePut
+			if t == reflect.TypeOf(v1.ObjectMeta{}) && jsonTag == "name" {
+				a.names = append(a.names, string(a.getData(vf, 0)))
+			}
+
+			// add actual object
+			if jsonTag != "" && jsonTag != "apiGroup" {
```

Comment:
nit: I generally would avoid huge if body for readability. Here, I think we can revert the condition here so the actual body is outside of if
```go
if jsonTag == "" || jsonTag == "apiGroup" {
    continue
}
b.Write(indent1)
b.WriteString(jsonTag)
b.WriteString("=")
...
```